feat: allow multiple filter values, fixes #7328

paulroub opened the pull request:
<!--
Thanks for submitting a pull request!

Please make sure you've read and understood our contributing guidelines here:
https://github.com/decaporg/decap-cms/blob/main/CONTRIBUTING.md

If this is a bug fix, make sure your description includes "fixes #xxxx", or
"closes #xxxx", where #xxxx is the issue number.

Please provide enough information so that others can review your pull request.
The first two fields are mandatory:
-->

**Summary**

Allows a collection filter to match one of a number of values; still allows the existing single-value syntax, as well as an array of values.

<!--
Explain the **motivation** for making this change.
What existing problem does the pull request solve?
-->

Need to manage a collection that's a *combination* of existing categories (rather than re-categorize all those posts).

**Test plan**

<!--
Demonstrate the code is solid.
Example: The exact commands you ran and their output, screenshots / videos if the pull request changes UI.
-->

- Added a unit test for the array variation, existing (non-array) tests still path.
- Tested against my collections, with single-value, single-element-array and multiple-element-array variations all working as expected.

**Checklist**

Please add a `x` inside each checkbox:

- [X] I have read the [contribution guidelines](https://github.com/decaporg/decap-cms/blob/main/CONTRIBUTING.md).
- [ ] I have added a documentation update (coming...)

**A picture of a cute animal (not mandatory but encouraged)**

![rp](https://github.com/user-attachments/assets/5eb66d81-35bd-4c34-9fe9-e232672497f5)

martinjagodic commented:
@paulroub, thanks for your contribution. Before we review, can you please ensure the tests are passing?

paulroub commented:
@martinjagodic Fixed the linting issues, along the way had to tell stylelint to ignore `backend.ts` as it was trying to parse a cast as CSS and was not happy with the result. No actual styles in this file, of course, so that should be harmless.

paulroub commented:
Rebased on current `main`, confirmed `npx run test:all` still succeeds.

paulroub commented:
> When I try this with my test config, the collection fails to load entries - even before I tried the list of filters. See screenshot below.
> 

I'm so far unable to reproduce this -- any chance you could share your test config?

paulroub commented:
> When I try this with my test config, the collection fails to load entries - even before I tried the list of filters. See screenshot below.
> 

Reproduced, fixed, added a test case. We were assuming the presence of the filtered-on field on all 
entries in the collection. Now just failing the match if (*e.g.*) we're filtering on "category" but a post doesn't have one.

> I also have an idea of extending this further by adding an operator like with the `condition` field.
> 

I had considered that as well, and poked at it a bit. I hadn't considered extending the syntax, that would make things much more doable. 

Perhaps a follow-on? This version seems useful on its own.

martinjagodic commented:
Great, thanks! We can do that in a later PR, I agree. If all is ok now, we should merge this for the 3.9 release.

martinjagodic reviewed (changes requested):
If I add an array as value in `dev-test.yaml` to the Posts collection like this

```yml
collections: # A list of collections the CMS should be able to edit
  - name: 'posts' # Used in routes, ie.: /admin/collections/:slug/edit
    filter: {field: title, value: ['This is post # 20']}
```

I should see one post entry, but I get these errors in console:

```
Warning: Failed prop type: Invalid prop `children` supplied to `ErrorBoundary`, expected a ReactNode. 

Warning: Failed prop type: Invalid prop `children` supplied to `Modal`, expected a ReactNode.

Uncaught (in promise) {type: 'ENTRIES_FAILURE', error: 'Failed to load entries', payload: 'TypeError: rawValue.toJS(...).toArray is not a function', meta: {…}}
```
